contracts/contracts/vault/VaultCore.sol

```diff
@@ -307,45 +292,31 @@ contract VaultCore is VaultStorage {
      *         strategies.
      * @return uint256 value Total value in USD (1e18)
      */
-    function totalValue() external returns (uint256 value) {
-        uint256[] memory assetPrices = _getAssetPrices(false);
-        value = _totalValue(assetPrices);
+    function totalValue() external view returns (uint256 value) {
+        value = _totalValue();
     }
 
     /**
      * @dev Internal Calculate the total value of the assets held by the
      *         vault and its strategies.
      * @return uint256 value Total value in USD (1e18)
      */
-    function _totalValue(uint256[] memory assetPrices)
-        internal
-        view
-        returns (uint256 value)
-    {
-        return
-            _totalValueInVault(assetPrices) +
-            _totalValueInStrategies(assetPrices);
+    function _totalValue() internal view returns (uint256 value) {
+        return _totalValueInVault() + _totalValueInStrategies();
```

Comment:
maybe won't happen anytime soon - but what happens when the vault is super successful and has so many assets values - then it overflows - or what if I take some massive flashloan on multiple assets and deposit - then cause overflow.

Reply:
I was curious about this, so I did some back of napkin math.

2^256 is kind of a big number. Conservatively it's 1e77. Let's use fixed point, 1e18, so we can represent 1e59 dollars. The current number of dollars in the world is around 1e13, so we are fairly good there. The current mass of the universe is estimated at 1e53 kilograms, which is 1e56 grams. I think we have enough bits available to store $1,000 for each gram of matter in the universe. Scaling ethereum's transaction per second might be a limiting factor here.

A bigger danger could be the compounding effects of yields. With year after year of 120% APY, (more than doubling every year) and the world using OUSD for its wealth, it could take 138 years for us run out of bits. I guess around year 136, we'd make a new contract with fewer decimals, and let people convert from one to the new OUSD.